plugins/woocommerce-admin/client/products/fills/details-section/details-section-fills.tsx

```diff
@@ -0,0 +1,124 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useSelect } from '@wordpress/data';
+import {
+	__experimentalWooProductSectionItem as WooProductSectionItem,
+	__experimentalWooProductFieldItem as WooProductFieldItem,
+	__experimentalProductFieldSection as ProductFieldSection,
+	useFormContext,
+} from '@woocommerce/components';
+import { registerPlugin } from '@wordpress/plugins';
+import { useState } from '@wordpress/element';
+import {
+	PRODUCTS_STORE_NAME,
+	WCDataSelector,
+	Product,
+} from '@woocommerce/data';
+
+/**
+ * Internal dependencies
+ */
+import { EditProductLinkModal } from '../../shared/edit-product-link-modal';
+import {
+	DetailsNameField,
+	DetailsCategoriesField,
+	DetailsFeatureField,
+	DetailsSummaryField,
+	DetailsDescriptionField,
+	DETAILS_SECTION_ID,
+} from './index';
+import './product-details-section.scss';
+
+const DetailsSection = () => {
+	const { values } = useFormContext< Product >();
+	const [ showProductLinkEditModal, setShowProductLinkEditModal ] =
+		useState( false );
+	const { permalinkPrefix, permalinkSuffix } = useSelect(
+		( select: WCDataSelector ) => {
+			const { getPermalinkParts } = select( PRODUCTS_STORE_NAME );
+			if ( values.id ) {
+				const parts = getPermalinkParts( values.id );
+				return {
+					permalinkPrefix: parts?.prefix,
+					permalinkSuffix: parts?.suffix,
+				};
+			}
+			return {};
+		}
+	);
+
+	return (
+		<>
+			<WooProductSectionItem
+				id={ DETAILS_SECTION_ID }
+				location="tab/general"
+				pluginId="core"
+			>
+				<ProductFieldSection
+					id="general/details"
+					title={ __( 'Product details', 'woocommerce' ) }
+					description={ __(
+						'This info will be displayed on the product page, category pages, social media, and search results.',
+						'woocommerce'
+					) }
+				/>
+			</WooProductSectionItem>
+			<WooProductFieldItem
+				id="details/name"
+				section={ DETAILS_SECTION_ID }
+				pluginId="core"
+			>
```

Comment:
Just thinking out loud:
The only part I dislike about Slotfills and this approach is having to individually wrap each field, it's really not a huge deal for now, but I wonder if we could clean this up a bit in the future (not a 100% sure yet how, but thought I mention it incase you have an idea).
But we can certainly leave this as is for now, as it does what its intended to do.

Reply:
That's actually bugged me as well, but it's hard to think of a solution that isn't fragile, cumbersome, or just doesn't do what we need it to. As it is users can always shove more than one field into a slot, but they will be targeted by a single ID and they'll need to manage the spacing/etc between fields. 

We could auto-generate IDs in some way, but that feels fragile and unintuitive. 🤷🏻